### PR TITLE
Fix InfoLayoutHelper not using first video stream, update details when selecting version

### PR DESCRIPTION
**Pull request by StubbyB**
On Android TV some tags are shown such as 'R', '4K' 'HEVC', etc.
Titles which have external subtitles, are missing the resolution and video codec tags (e.g. '1080' and 'H264').

The source of the issue is `org.jellyfin.androidtv.util.InfoLayoutHelper.addRatingAndRes` getting all media streams, and assuming that the video stream will be the first one, when in fact external subtitles are first (also verifiable by looking at the debug info at the bottom).

**Changes**
Instead of assuming the video is the first stream, take the first stream tagged as `MediaStreamType.VIDEO`.

**Issues**
#2811

### CODE REVIEW

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]
We already have a `StreamHelper.getFirstAudioStream` function so it would probably make sense to add a `StreamHelper.getFirstVideoStream` or maybe even merge them to a `StreamHelper.getFirstStreamOfType()`.

**Comment by StubbyB** [comment]
> We already have a `StreamHelper.getFirstAudioStream` function so it would probably make sense to add a `StreamHelper.getFirstVideoStream` or maybe even merge them to a `StreamHelper.getFirstStreamOfType()`.

Moved logic to `StreamHelper`.
Also, since the order of the video versions in the selection menu is the order of the media sources, I added the functionality of settings the tags given a selected media source index, with 0 being the default (preserving existing behavior).
Had to add function signatures accordingly.

**Comment by StubbyB** [comment]
made a few cosmetic changes, done modifying until reviewed :) @nielsvanvelzen

File: app/src/main/java/org/jellyfin/androidtv/ui/itemdetail/MyDetailsOverviewRow.kt [hunk 1]
```diff
@@ -14,6 +14,7 @@ class MyDetailsOverviewRow @JvmOverloads constructor(
 	var infoItem1: InfoItem? = null,
 	var infoItem2: InfoItem? = null,
 	var infoItem3: InfoItem? = null,
+	var selectedMediaSourceIndex: Int = 0
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/ui/itemdetail/MyDetailsOverviewRow.kt:17]:
```suggestion
	var selectedMediaSourceIndex: Int = 0,
```

File: app/src/main/java/org/jellyfin/androidtv/util/InfoLayoutHelper.java [hunk 2]
```diff
@@ -45,17 +45,20 @@ public static void addInfoRow(Context context, BaseRowItem item, LinearLayout la
                 break;
         }
     }
-
-    public static void addInfoRow(Context context, BaseItemDto item, LinearLayout layout, boolean includeRuntime, boolean includeEndTime) {
+    public static void addInfoRow(Context context, BaseItemDto item, int mediaSourceIndex, LinearLayout layout, boolean includeRuntime, boolean includeEndTime) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/InfoLayoutHelper.java:48]: (missing the empty line before the function)
```suggestion

    public static void addInfoRow(Context context, BaseItemDto item, int mediaSourceIndex, LinearLayout layout, boolean includeRuntime, boolean includeEndTime) {
```

File: app/src/main/java/org/jellyfin/androidtv/util/InfoLayoutHelper.java [hunk 3]
```diff
@@ -325,14 +328,17 @@ private static void addDate(@NonNull Context context, BaseItemDto item, LinearLa
...
             addBlockText(context, layout, item.getOfficialRating());
             addSpacer(context, layout, "  ");
         }
-        if (item.getMediaStreams() != null && item.getMediaStreams().size() > 0 && item.getMediaStreams().get(0).getWidth() != null && item.getMediaStreams().get(0).getHeight() != null) {
-            int width = item.getMediaStreams().get(0).getWidth();
-            int height = item.getMediaStreams().get(0).getHeight();
+
+        MediaStream videoStream = StreamHelper.getFirstVideoStream(item, mediaSourceIndex);
+
+        if(videoStream != null && videoStream.getWidth() != null && videoStream.getHeight() != null) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/InfoLayoutHelper.java:339]:
```suggestion
        if (videoStream != null && videoStream.getWidth() != null && videoStream.getHeight() != null) {
```

File: app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java [hunk 4]
```diff
@@ -26,9 +26,33 @@ public static List<MediaStream> getAudioStreams(MediaSourceInfo mediaSource) {
...
+        return getStreams(mediaSource, MediaStreamType.VIDEO);
+    }
+
     public static MediaStream getFirstAudioStream(BaseItemDto item) {
-        if (item.getMediaSources() == null || item.getMediaSources().size() < 1) return null;
-        List<MediaStream> streams = getAudioStreams(item.getMediaSources().get(0));
+        return getFirstAudioStream(item, 0);
+    }
+
+    public static MediaStream getFirstAudioStream(BaseItemDto item, int mediaSourceIndex) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java:37]:
```suggestion
    public static MediaStream getAudioStream(BaseItemDto item, int mediaSourceIndex) {
```
It's not the first if we explicitly tell it the index.

File: app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java [hunk 5]
```diff
@@ -26,9 +26,33 @@ public static List<MediaStream> getAudioStreams(MediaSourceInfo mediaSource) {
...
+        return getFirstStreamOfType(item, MediaStreamType.AUDIO, mediaSourceIndex);
+    }
+
+    public static MediaStream getFirstVideoStream(BaseItemDto item) {
+        return getFirstVideoStream(item, 0);
+    }
+
+    public static MediaStream getFirstVideoStream(BaseItemDto item, int mediaSourceIndex) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java:45]:
```suggestion
    public static MediaStream getVideoStream(BaseItemDto item, int mediaSourceIndex) {
```

File: app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java [hunk 6]
```diff
@@ -26,9 +26,33 @@ public static List<MediaStream> getAudioStreams(MediaSourceInfo mediaSource) {
...
+        return getFirstStreamOfType(item, MediaStreamType.VIDEO, mediaSourceIndex);
+    }
+
+    public static MediaStream getFirstStreamOfType(BaseItemDto item, MediaStreamType streamType) {
+        return getFirstStreamOfType(item, streamType, 0);
+    }
+
+    public static MediaStream getFirstStreamOfType(BaseItemDto item, MediaStreamType streamType, int mediaSourceIndex) {
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java:53]:
```suggestion
    public static MediaStream getStreamOfType(BaseItemDto item, MediaStreamType streamType, int mediaSourceIndex) {
```

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java (hunk 4 above)
Comment (StubbyB) [reply app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java:37]: It is the first matching `MediaStream` for the given `MediaSource`.

The idea was to allow the existing functionality of retrieving the first overall media stream, and also an overload allowing to choose a specific source.

Also `getFirstAudioStream` is an existing public util which I was reluctant to change.

Would you rather it be a separate function signature named something like:
`getFirstStreamOfSource(BaseItemDto item, int mediaSourceIndex)`
`getFirstStreamOfSource(BaseItemDto item, int mediaSourceIndex, MediaStreamType streamType)`?

Comment (StubbyB) [reply app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java:37]: This applies to all other comments ofc

Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/util/apiclient/StreamHelper.java:37]: Aah you're right, I missed that part. This naming is fine 👍.

**Review by nielsvanvelzen — APPROVED** [review]

**Comment by nielsvanvelzen** [comment]
Thank you!

**Pull request merged**